maps/game_map.go

```diff
@@ -18,13 +18,68 @@ type GameMap interface {
 	UpdateBoard(previousBoardState *rules.BoardState, settings rules.Settings, editor Editor) error
 }
 
+// dimensions is an internal type used to encapsulate the configuration of a particular map size.
+type dimensions struct {
+	// Width is the width, in number of map squares, of the map.
+	// The value 0 has a special meaning to mean unlimited.
+	Width uint
+	// Height is the height, in number of map squares, of the map.
+	// The value 0 has a special meaning to mean unlimited.
+	Height uint
+}
+
+// sizes is an interal type used to encapsulate the configuration of supported map sizes.
+type sizes []dimensions
+
+// IsUnlimited reports whether the supported sizes are unlimited.
+// Note that even for unlimited sizes, there will be an upper bound that can actually be run and visualised.
+func (d sizes) IsUnlimited() bool {
+	return len(d) == 1 && d[0].Width == 0
+}
+
+// UnlimitedSizes creates sizes for a map that has no fixed sizes (supports unlimited sizes).
+func UnlimitedSizes() sizes {
```

Comment:
What do you think of using `AnySize` in place of `UnlimitedSizes`? Just from a naming perspective.

Reply:
I was also thinking that, rather than have `0` be a special value, what if we used no sizes specified to indicate any size is okay?

```len(Meta.Sizes) == 0```

Reply:
Yeah, I like `AnySize` much better :+1:

>  was also thinking that, rather than have 0 be a special value, what if we used no sizes specified to indicate any size is okay?

Yeah I considered that but I wanted to avoid having the zero value so it would have to be explicit. But that being said, I don't feel super strongly about it. It does seem a bit cleaner than having to check the first dimensions for a 0 width.

Reply:
I'd have to remove the registry test that ensures dimensions are specified (being able to test that was part of the reason I went with this approach).

Reply:
Yeah, you're right in that also validates the zero value which is gross. I'm good with what you've got, shrug.